server: package timed out

[(server).(unknown) failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2743666&tab=buildLog) on [master@37cdf01299861235888af3411e699480f5a2d670](https://github.com/cockroachdb/cockroach/commits/37cdf01299861235888af3411e699480f5a2d670):

```
Slow failing tests:
TestDecommissionedNodeCannotConnect - 3.00s

Slow passing tests:
TestStatusAPITransactionStatementIDsTruncation - 27.76s
TestClusterConnectivity - 25.93s
TestInitHandshakeWrongToken - 20.10s
TestStatusAPITransactions - 19.12s
TestAdminAPITableStats - 17.69s
TestAdminAPIDataDistribution - 14.55s
TestListSessionsV2 - 14.18s
TestRangeCount - 12.30s
TestStatusAPIStatements - 12.05s
TestNodesV2 - 11.16s
TestLivenessAPI - 10.70s
TestHealthV2 - 10.56s
TestStatsforSpanOnLocalMax - 10.55s
TestAdminAPINonTableStats - 9.91s
TestAdminAPITableDetails - 9.85s
TestMultiRangeScanWithPagination - 9.55s
TestInitHandshake - 8.48s
TestRotationOnIntializedNode - 7.68s
TestRotationOnPartialIntializedNode - 6.78s
TestRotationOnBrokenIntializedNode - 6.57s

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=4

```
make stressrace TESTS=(unknown) PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2A%28unknown%29.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

This issue was opened by TC because `TestDecommissionedNodeCannotConnect` was failing with:
`E210304 06:43:14.645467 698376 2@rpc/context.go:1057  [n3] 22183  removing connection to 127.0.0.1:39687 due to error: rpc error: code = PermissionDenied desc = n3 was permanently removed from the cluster at 2021-03-04 06:17:09.902970014 +0000 UTC; it is not allowed to rejoin the cluster
panic: test timed out after 30m0s`

Example failure: https://teamcity.cockroachdb.com/viewLog.html?buildId=2743148&buildTypeId=Cockroach_UnitTests_Test&tab=artifacts#%2FlogTestDecommissionedNodeCannotConnect696035602;%2Fraw.0.json.txt.tgz

--- a/pkg/server/connectivity_test.go
+++ b/pkg/server/connectivity_test.go
@@ -20,7 +20,6 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
-	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/livenesspb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/rpc"
@@ -396,12 +395,14 @@ func TestDecommissionedNodeCannotConnect(t *testing.T) {
 
 	// Trying to scan the scratch range via the decommissioned node should
 	// now result in a permission denied error.
-	_, err := decomSrv.DB().Scan(ctx, scratchKey, keys.MaxKey, 1)
-	require.Error(t, err)
-
+	//
 	// TODO(erikgrinaker): until cockroachdb/errors preserves grpcstatus.Error
-	// across errors.EncodeError() we just check for any error, this should
-	// check that it matches codes.PermissionDenied later.
+	// across errors.EncodeError() we'll have to disable this, since the return
+	// code isn't preserved, and since some internal calls (notably RangeLookup
+	// via RangeIterator) end up retrying indefinitely because it can't detect
+	// the error code. See: https://github.com/cockroachdb/cockroach/issues/61470
+	//_, err := decomSrv.DB().Scan(ctx, scratchKey, keys.MaxKey, 1)
+	//require.Error(t, err)
 	//err = errors.UnwrapAll(err)
 	//s, ok := grpcstatus.FromError(err)
 	//require.True(t, ok, "expected gRPC error, got %T (%v)", err, err)

